### PR TITLE
suggestions for 2024 roadmap

### PR DESCRIPTION
**Pull request by generall**
Suggestion for 2024 roadmap.

Most likely we will come up with some new unexpected stuff anyway, but general direction is this.

Please feel free to comment and suggest

### CODE REVIEW

**Review by timvisee — APPROVED** [review]
Super excited! Let's bring vector search to a whole different level in 2024. :rocket:

File: docs/roadmap/README.md [hunk 1]
```diff
@@ -26,34 +34,39 @@ For those familiar with Rust - check out our [contribution guide](https://github
...
-* :arrow_double_up: Scalability
-  * [ ] Automatic replication factor adjustment
-  * [ ] Automatic shard distribution on cluster scaling
-  * [ ] Repartitioning support
+* 🏗️ Scalability
+  * [ ] Faster shard synchronization
+    * [ ] Non-blocking snapshotting
+    * [ ] Incremental replication
+  * [ ] Specialized nodes
+    * [ ] Read-only nodes
```
Comment (NirantK) [inline docs/roadmap/README.md:52]: Ooh, read only nodes would be amazing for high RPS!

File: docs/roadmap/README.md [hunk 2]
```diff
@@ -26,34 +34,39 @@ For those familiar with Rust - check out our [contribution guide](https://github
...
-## Milestones
+## Core Milestones
 
-* :atom_symbol: Quantization support
-  * [x] Scalar quantization f32 -> u8 (4x compression)
-  * [x] Product quantization (4x, 8x, 16x, 32x, and 64x compression)
-  * [x] Binary quantization (32x compression, 40x speedup)
-  * [ ] Support for binary vectors
+* 📃 Hybrid Search and Sparse Vectors
+  * [ ] Make Sparse Vectors serving as cheap and fast as Dense Vectors
```
Comment (NirantK) [inline docs/roadmap/README.md:40]: Woohoo @agourlay !

File: docs/roadmap/README.md [hunk 3]
```diff
@@ -26,34 +34,39 @@ For those familiar with Rust - check out our [contribution guide](https://github
...
+  * [ ] Introduce Hybrid Search into Qdrant Client
+    * [ ] Dense + Sparse + Fusion in one request
+    * [ ] Customizable Re-Ranking
```
Comment (NirantK) [inline docs/roadmap/README.md:43]: Do we've a rough sense of what will be allow folks to tune in the re-ranking?

File: docs/roadmap/README.md [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
   * Storage should be compatible between any two consequent versions, so you can upgrade Qdrant with automatic data migration between consecutive versions.
-* **Make billion-scale serving cheap** - qdrant already can serve billions of vectors, but we want to make it even more affordable.
-* **Easy scaling** - our plan is to make it easy to dynamically scale Qdrant, so you could go from 1 to 1B vectors seamlessly.
-* **Various similarity search scenarios** - we want to support more similarity search scenarios, e.g. sparse search, grouping requests, diverse search, etc.
+* **Make serving easy on multi-billion scale** - qdrant already can can serve billions of vectors cheaply, using such techniques as quantization. In the 2024 year, we plan to make it even easier to scale Qdrant.
+  * Faster and more reliable replications
+  * Out-of-the-box read-write segregation
+  * Specialized nodes and multi-region deployments
+* **Better ecosystem** - in 2023 we introduced [fastembed](https://github.com/qdrant/fastembed) to simplify embedding generation but keep it out of the core.
```
Comment (NirantK) [inline docs/roadmap/README.md:19]: Glad to see this! <3

**Review by NirantK — COMMENTED** [review]
Super exciting stuff! Let's goooo!

File: docs/roadmap/README.md [hunk 5]
```diff
@@ -26,34 +34,39 @@ For those familiar with Rust - check out our [contribution guide](https://github
...
 ---
 
-* Additionally
-  * [ ] Extend full-text filtering support
-    * [ ] Support for phrase queries
-    * [ ] Support for logical operators
-  * [x] Simplify update of collection parameters
+* 🏝️ New Data Exploration techniques
+  * [ ] Improvements in Discovery API to support more use-cases
+  * [ ] Diversity Search
```
Comment (coszio) [inline docs/roadmap/README.md:68]:
```suggestion
  * [ ] Diversity Sampling
```

File: docs/roadmap/README.md [hunk 6]
```diff
@@ -1,19 +1,27 @@
...
 Goals of the release:
 
-* **Maintain easy upgrades** - we plan to keep backward compatibility for at least one minor version back.
+* **Maintain easy upgrades** - we plan to keep backward compatibility for at least one minor version back (this stays the same in 2024).
   * That means that you can upgrade Qdrant without any downtime and without any changes in your client code within one minor version.
   * Storage should be compatible between any two consequent versions, so you can upgrade Qdrant with automatic data migration between consecutive versions.
-* **Make billion-scale serving cheap** - qdrant already can serve billions of vectors, but we want to make it even more affordable.
-* **Easy scaling** - our plan is to make it easy to dynamically scale Qdrant, so you could go from 1 to 1B vectors seamlessly.
-* **Various similarity search scenarios** - we want to support more similarity search scenarios, e.g. sparse search, grouping requests, diverse search, etc.
+* **Make serving easy on multi-billion scale** - qdrant already can can serve billions of vectors cheaply, using such techniques as quantization. In the 2024 year, we plan to make it even easier to scale Qdrant.
```
Comment (coszio) [inline docs/roadmap/README.md:15]:
```suggestion
* **Make serving easy on multi-billion scale** - Qdrant already can serve billions of vectors cheaply, using techniques as quantization. In the 2024 year, we plan to make it even easier to scale it.
```

File: docs/roadmap/README.md (hunk 3 above)
Comment (generall) [reply docs/roadmap/README.md:43]: i was thinking about cross-encoders for the beginning. Especially If we can ship them pytorch-free

File: docs/roadmap/README.md [hunk 7]
```diff
@@ -1,19 +1,27 @@
...
+  * Faster and more reliable replications
+  * Out-of-the-box read-write segregation
+  * Specialized nodes and multi-region deployments
+* **Better ecosystem** - in 2023 we introduced [fastembed](https://github.com/qdrant/fastembed) to simplify embedding generation but keep it out of the core.
+  In 2024 we plan to continue this trend: implement more advanced and specialized tools while keeping the core focused on the main use-case.
+  * Advanced support for sparse vectors - we plan to make sparse vectors inference as fast and easy as the dense one.
+  * Hybrid search out of the box with no overhead - something you can build with Qdrant today, but in a more convenient way.
```
[the remark this answers is not in the document]
Comment (generall) [reply docs/roadmap/README.md:22]: We would still need to decide about the interface, but that is the idea

**Review by agourlay — APPROVED** [review]

**Pull request merged**